opencart1.5.x/upload/catalog/controller/payment/xenditcc.php

```diff
@@ -1,286 +1,16 @@
-        }
-        
-        $this->cart->clear();
-
-        $this->model_payment_xendit->paidOrder($order_id, $charge['created'], array('xendit_charge_id' => $charge['id']));
-
-        $message = 'Payment successful. Charge ID: ' . $charge['id'];
-        $this->model_checkout_order->update(
-            $order_id,
-            2,
-            $message,
-            false
-        );
-
-        $redir_url = $this->url->link('checkout/success');
-        $this->response->redirect($redir_url);
-    }
-
-    private function cancel_order($order_id, $message) {
-        $this->model_payment_xendit->cancelOrder($order_id);
-        $this->model_checkout_order->update(
-            $order_id,
-            7,
-            $message,
-            false
-        );
-
-        $redir_url = $this->url->link('payment/xenditcc/failure');
-        $this->response->redirect($redir_url);
-        return;
-    }
-
-    private function get_api_key() {
-        if ($this->config->get('payment_xendit_environment') === 'live') {
-            return array(
-                'secret_key' => $this->config->get('payment_xendit_live_secret_key'),
-                'public_key' => $this->config->get('payment_xendit_live_public_key')
-            );
-        } else {
-            return array(
-                'secret_key' => $this->config->get('payment_xendit_test_secret_key'),
-                'public_key' => $this->config->get('payment_xendit_test_public_key')
-            );
-        }
-    }
-
-    private function map_failure_reason($failure_reason) {
-        $card_declined_reason = 'Card declined by the issuer bank. Please try with another card or contact the bank directly.';
-        switch ($failure_reason) {
-            case 'CARD_DECLINED':
-                return $card_declined_reason . ' Code: 200011';
-            case 'STOLEN_CARD':
-                return $card_declined_reason . ' Code: 200013';
-            case 'INSUFFICIENT_BALANCE':
-                return 'Card declined due to insufficient balance. Ensure sufficient balance is available, or try another card. Code: 200012';
-            case 'INVALID_CVN':
-                return 'Card declined due to incorrect card details. Please try again. Code: 200015';
-            case 'INACTIVE_CARD':
-                return $card_declined_reason . ' Code: 200014';
-            case 'EXPIRED_CARD':
-                return 'Card declined due to expiration. Please try again with another card. Code: 200010';
-            case 'PROCESSOR_ERROR':
-                return 'We encountered an issue while processing the checkout. Please try again. Code: 200009';
-            case 'AUTHENTICATION_FAILED':
-                return 'The authentication process failed. Please try again. Code: 200001';
-            case 'UNEXPECTED_PLUGIN_ISSUE':
-                return 'We encountered an issue processing your checkout, please contact us. Code: 100007';
-            default: return $failure_reason;
-        }
-    }
```

Comment:
that's a lot of deleted line, just want to check this is intended?

Reply:
Yes, the old implementation is deprecated and we want just to utilized the Xeninvoice (so that why this task named cards migration) endpoint for credit card. The xeninvoice itself is integrated with all xendit payment through Invoice endpoint `/payment/xendit/invoice`